benchmarks/showcases/LazyGridImageView/nativeAndroidApp/app/src/main/java/org/jetbrains/lazygridimage/MainActivity.kt

```diff
@@ -0,0 +1,162 @@
+        List(999) { index ->
+            val imageFilenames = context.assets.list("drawable") ?: emptyArray()
+            "drawable/${imageFilenames[index % imageFilenames.size]}"
+        }
+    }
+
+    val numOfColumns = 3
+
+    // Auto-scrolling effect
+    LaunchedEffect(autoScroll) {
+        if (autoScroll) {
+            while (true) {
+                delay(100)
+
+                if (scrollingDown) {
+                    if (currentIndex < drawableResources.size - numOfColumns - 1) {
+                        currentIndex += numOfColumns
+                    } else {
+                        scrollingDown = false
+                    }
+                } else {
+                    if (currentIndex > 0) {
+                        currentIndex -= numOfColumns
+                    } else {
+                        scrollingDown = true
+                    }
+                }
+
+                coroutineScope.launch {
+                    gridState.animateScrollToItem(
+                        index = currentIndex,
+                    )
+                }
+            }
+        }
+    }
+
+    MaterialTheme {
+        Surface(
+            modifier = Modifier.fillMaxSize(),
+            color = MaterialTheme.colorScheme.background
+        ) {
+            Column(
+                modifier = Modifier.fillMaxSize(),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                // Checkbox for auto-scrolling
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier.padding(8.dp)
+                ) {
+                    Checkbox(
+                        checked = autoScroll,
+                        onCheckedChange = { autoScroll = it }
+                    )
+                    Spacer(modifier = Modifier.width(8.dp))
+                    Text("Auto Scroll")
+                }
+
+                LazyVerticalGrid(
+                    columns = GridCells.Fixed(numOfColumns),
+                    contentPadding = PaddingValues(4.dp),
+                    modifier = Modifier.fillMaxSize(),
+                    state = gridState
+                ) {
+                    items(drawableResources) { imagePath ->
+                        ImageCard(
+                            imagePath = imagePath,
+                            modifier = Modifier.padding(4.dp)
+                        )
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun ImageCard(
+    imagePath: String,
+    modifier: Modifier = Modifier
+) {
+    Card(
+        modifier = modifier
+            .aspectRatio(1f) // Square aspect ratio
+            .fillMaxWidth(),
+        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
+    ) {
+        val context = LocalContext.current
+        context.assets.open(imagePath).use { inputStream ->
+            val bitmap = BitmapFactory.decodeStream(inputStream)
```

Comment:
Did you consider using `painterResource(Int)` instead? It will be a closer analog to the CMP resources.

Ids can be found with:
```
context.resources.getIdentifier("name", "drawable", context.packageName)
```

Reply:
But there is no way to get all drawable ids, right? CMP resources are stored in Android assets, so ideologically CMP resources are closer to Android ones, but in the sense of comparison assets may be better to use.

Reply:
> CMP resources are stored in Android assets

It looks like implementation details. For an Android developer resources != assets

> But there is no way to get all drawable ids

We can hardcode "image$num"

Reply:
Ok, I rewrote it as you suggest